### PR TITLE
Add test pipeline

### LINKED ISSUE #12: Add Realtime Trains extract test
### Task

- [x] Add test to Realtime Trains

### User Story

As a developer, I want to test my code so I can be sure that it works as intended

### PR DESCRIPTION
**Pull request by Zhi-704**
Add test for all files in the pipeline, tests for cases where there is an API error, different expected input type, as well as when there is no data being retrieved. Close #12 #38

### CODE REVIEW

**Comment by Zhi-704** [comment]
Annoyingly, if anyone uses the same test file name even though it's in a different directory the checks will fails. So have renamed to avoid that error

File: .util/pytest_scores.json [hunk 1]
```diff
@@ -1 +1 @@
-{"passed": 26, "failed": 0, "total": 26, "passed_percentage": 100}
\ No newline at end of file
+{"passed": 58, "failed": 0, "total": 58, "passed_percentage": 100}
```
Comment (fm1psy) [inline .util/pytest_scores.json:1]: good job writing all these tests

File: realtime_trains/test_extract_real.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+from requests.exceptions import RequestException
+from requests.auth import HTTPBasicAuth
+from requests import Response
+
+from extract_real import (
+    get_data_from_api,
+    get_yesterday_data_of_station,
+    get_all_stations_crs,
+    get_api_data_of_all_stations
+)
```
Comment (fm1psy) [inline realtime_trains/test_extract_real.py:15]: did not know you could use brackets in an import statement, I'm impressed

**Review by fm1psy — APPROVED** [review]
Your tests are very detailed, and the fact they all pass gives me confidence in the strength of your code. Good job

File: realtime_trains/test_extract_real.py
Comment (A-Faris) [inline realtime_trains/test_extract_real.py:1]: Thank you for testing

**Pull request merged**